### PR TITLE
basics-10-2 - therese lau

### PR DESCRIPTION
**Pull request by thereseLYR**
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?
3

Please fill in one error and/or error message you received while working on this assignment.
used `.tostring` instead of `.toString`

What part of the assignment did you spend the most time on?
figuring out how to design the initial release in a way that wouldn't force me to refactor large swathes of logic, e.g. using nested arrays as a scorekeeping mechanism so i can autogenerate template arrays for multiple players or dice for the MC stuff.

Comfort Level (1-5):
4 {without MC)

Completeness Level (1-5):
4 {without MC}

What did you think of this deliverable?
pretty big list of MC requirements! i'll probably want to learn how to use switch statements along the way.

Is there anything in this code that you feel pleased about?
my main function seems neat :^)

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -1,4 +1,183 @@
...
+    // https://stackoverflow.com/questions/1069666/sorting-object-property-by-values
+  );
+  console.log(`sorted: ${sortedLeaderboardObject}`);
+  return sortedLeaderboardObject;
+};
+
+var generateLeaderboardMessage = function (obj) {
+  var messageString = "Leaderboard: <br><br>";
+  for (let [key, value] of Object.entries(obj)) {
+    messageString = messageString + `Player ${key}:   ${value}` + "<br><br>";
```
[the remark this answers is not in the document]
Comment (thereseLYR) [reply script.js]: ok i could have made this a new line with <br> in the template literal

File: README.md [hunk 2]
```diff
@@ -1 +1,18 @@
...
+
+<ul> Game Mode Modifiers
+  <li> Lowest Combined Mode			(--lowestCombined)</li>
+  <li> Auto-Generate Mode 			(--auto)</li>
+  <li> Variable Number of Dice 		(--vDiceN)</li>
+</ul>
+
+Modifiers must be applied in the start line with their respective flag, as indicated in the parentheses above. Multiple modifiers can be applied in each game, but the Variable Number of Dice modifier only works with the Auto-Generate modifier enabled. Similar to command-line programs such as [wget](https://www.gnu.org/software/wget/), all modifiers must be declared at the same time. 
+
+An example of a working start command would be `start --auto --vDice6`.
```
Comment (JustinWong98) [inline README.md:18]: Why do you have to flex so hard ;-;

File: script.js [hunk 3]
```diff
@@ -1,4 +1,324 @@
...
+// variables for MVP- 2 player manual
+var activePlayer = 1;
+var currentGameProgressMode = "initializing";
+var currentPlayerArray = [0, 0];
+var currentScore = 0;
+var scoreArray = [[], []];
+// ^this is meant to be a nested array - index 0 corresponds to player 1's scores, etc.
+// initialize later in main
+var progressNumber = 0; // incremented through player turns until a round is complete
+const validCommands = ["start", "Roll", "dice 1", "dice 2", "next"];
```
Comment (JustinWong98) [inline script.js:19]: Wow interesting use of an array to store commands

File: script.js [hunk 4]
```diff
@@ -1,4 +1,324 @@
...
+];
+
+var validateInput = function (input) {
+  var parsedInput = "";
+  if (input.includes("dice") || input.includes(" ") == false) {
+    parsedInput = input;
+  } else if (input.includes(" ") == true) {
+    parsedInput = input.substr(0, input.indexOf(" "));
+    // check only substring before the first space
+  }
```
Comment (JustinWong98) [inline script.js:42]: Impressive!

File: script.js [hunk 5]
```diff
@@ -1,4 +1,324 @@
...
+  console.log(`removed ${removedElement}`);
+  return arr;
+};
+
+var parseVariableDiceModifier = function (vdicemdf) {
+  // vdicemdf should take the form "--vdiceX", where X is an integer
+  // the "--vdice" substring occupies the 0th position to the 6th position
+  var customNumberOfDice = vdicemdf.substring(7); // remove "--vdice" and read remaining numbers
+  customNumberOfDice = Number(customNumberOfDice);
+  numberOfDice = customNumberOfDice;
```
Comment (JustinWong98) [inline script.js:64]: Is it possible to have just done numberOfDice = Number(customNumberOfDice) ?

**Review by JustinWong98 — COMMENTED** [review]
Well, I think I'll still need a bit of time to look through and fully digest the code, I'll play around with the app and see if there's any changes I would make. So far, I think that you could probably refactor the main function to avoid too many if statements, but ill make another review if I can think of a way to do so

But overall, pretty sure this is the most unique beat that project so far LOL

File: script.js [hunk 6]
```diff
@@ -1,4 +1,76 @@
...
+    return "Hello Player 1. Enter 'Roll' to continue.";
+  }
+  if (input == "Roll") {
+    return playerTurnRoll(activePlayer);
+  }
+  if (input == "dice 1" || input == "dice 2") {
+    playerTurnSelection(input);
+    writePlayerScore();
+    return switchPlayer();
+  } else {
```
[the remark this answers is not in the document]
Comment (JustinWong98) [reply script.js]: If all of the conditionals have a return statement in them, then yeah there's no need to actually do an else statement because by default if the program reaches past line 73 then that means none of the previous statements were true, meaning that it is an else by default

File: README.md (hunk 2 above)
Comment (thereseLYR) [reply README.md:18]: i wanted to use modifier flags to minimize the changes i would have to make to the `main` function - if i had introduced multiple pages of instructions, that would probably make my main function longer and messier than it already was. modifier flags in the input string allowed me to minimize changes to `main`, and shove everything into a helper function
TLDR i didnt want to change main that much lol

File: script.js (hunk 5 above)
Comment (thereseLYR) [reply script.js:64]: yes ;-;
